src/pages/index.js

```diff
@@ -124,4 +143,150 @@ const enableValidation = (config) => {
+        formValidators["profile-form"].resetFormValidation();
+      }, 1500)
+    });
+  },
+});
+
+const addPopup = new PopupWithForm(addPhotoPopup, {//new hotness
+  handleFormSubmit: ({ title, image_url: imageUrl }) => {
+    const newObj = {
+      name: title,
+      link: imageUrl
+    };
+    api.postCard(newObj)
+    .then((data) => {
+      renderCard(data);
+      console.log("IT GOT THIS FAR.");
+      addPopup.renderLoading(true, "Creating and adding card...");
+      formValidators["add-photo-form"].disableSubmitButton();
+    })    
+    .then(() => {
+      //addPopup.close();
+      setTimeout(() => {
+        addPopup.close();
+        formValidators["add-photo-form"].resetFormValidation();
+        formValidators["add-photo-form"].clearInputs();
+        addPopup.renderLoading(false);
+      }, 1500);
+    })
+    .catch((err) => {
+      addPopup.renderLoading(true, "Error while creating card! Please wait and try again.");
+      console.log(err);
+      setTimeout(() => {
+        addPopup.renderLoading(false);
+        formValidators["add-photo-form"].resetFormValidation();
+      }, 1500)
+    })
+  },
+});
+
+const cardSection = new Section({items: starterPack, renderer: renderCard}, ".cards");//I know the array is still empty but I fill it later
+
+const avatarPopup = new PopupWithForm(editAvatarPopup, {handleFormSubmit: ({avatarUrl}) => {//folding into api model
+  api.patchAvatar({avatar: avatarUrl})
+  .then(() => {
+    avatarPopup.renderLoading(true, "Changing avatar...");
+    user.setAvatar(avatarUrl);
+    formValidators["avatar-form"].disableSubmitButton();
+  })
+  .then(() => {
+    setTimeout(() => {
+      formValidators["avatar-form"].resetFormValidation();
+      avatarPopup.close();
+      avatarPopup.renderLoading(false);
+      formValidators["avatar-form"].clearInputs();
+    }, 1500);
+  })
+  .catch((err) => {
+    console.log(err);
+    avatarPopup.renderLoading(true, "Error while changing avatar! Please wait and try again");
+    setTimeout(() => {
+      formValidators["avatar-form"].resetFormValidation();
+      avatarPopup.renderLoading(false);
+    }, 1500);
+  });
+}});
+
+const deleteDialogue = new PopupFormNoFields(".delete-popup", {handleFormSubmit: handleDeleteSubmit});//"oh hey mike why didn't you just make a function to pass into the other constructors?" Because I was having such an abysmal time with figuring out how to delete, I'm leaving the others as is, don't care if it looks messy
+
+const imageBox = new PopupWithImage(cardPreviewPopup);
+
+//calls
+
+//getting the avatar image loaded
+api.getUserInfo()
+.then((data) => {
+  user.setAvatar(data.avatar);
+})
+.catch((err) => {
+  console.log("Unable to load avatar image! Please wait and try again. " + err);
+});
+
+//loading up cards into the array that the Card section draws from and then rendering them.
+api.getCards()
+.then((data) => {
+  starterPack.push(data);
+  cardSection.renderItems();
```

Comment:
**Could be improved**

You can change it like this to make the code reusable for any arrays in the future

```js

.then((data) => {
  cardSection.renderItems(data);
```